web/web/models/notification.py

```diff
@@ -1,29 +1,104 @@
-from marshmallow import Schema, fields
+from itertools import chain
+from sqlalchemy import text, func
+from sqlalchemy.sql import label
+from sqlalchemy.types import TIMESTAMP
+from sqlalchemy.schema import UniqueConstraint
+from marshmallow import fields
 from marshmallow_sqlalchemy import SQLAlchemyAutoSchema
-# from sqlalchemy.types import TIMESTAMP
-# from web.database import (
-#     db,
-#     Model,
-#     Column,
-#     SurrogatePK,
-#     relationship,
-#     reference_col,
-# )
 
-# class Notification(Model):
-#     __tablename__ = 'notification'
+from web.database import (
+    db,
+    Model,
+    Column,
+    SurrogatePK,
+    relationship,
+    reference_col,
+)
 
-# def __repr__(self):
-#     return f'<Notification>'
+class Notification(Model):
+    __tablename__ = 'notifications'
 
+    notification_id = Column(db.Integer, 
+                             primary_key=True,
+                             autoincrement=True, 
+                             nullable=False)
+    
+    alert_type_id = Column(db.Integer,
+                           db.ForeignKey('alert_types.alert_type_id'),  
+                           nullable=False)
 
-class NotificationTypeSchema(Schema):
-    notification_type = fields.Str()
-    label = fields.Str()
-    is_active = fields.Bool()
+    email = Column(db.String(255),
+                   nullable=False)
 
+    is_active = Column(db.Boolean,
+                       default=False,
+                       nullable=False)
 
-class NotificationSchema(Schema):
-    pk = fields.Str()
-    email = fields.Str()
-    notifications = fields.List(fields.Nested(NotificationTypeSchema))
+    created_by = Column(db.Integer,
+                        db.ForeignKey('users.id'),
+                        nullable=False)
+
+    updated_at = Column(TIMESTAMP, 
+                        nullable=False,
+                        server_default=text('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP'))
+
+    created_at = Column(TIMESTAMP,
+                        nullable=False,
+                        server_default=func.now())
+
+    # Unique constraint for alert_type_id and email
+    __table_args__ = (UniqueConstraint('alert_type_id', 'email', name='_alert_type_id_email_uc'),
+                     )
+
+    # Methods
+    def __repr__(self):
+        return f'<Notification notification_id={self.notification_id} alert_type_id={self.alert_type_id} is_active={self.is_active}>'
+
+    def get_all_notifications_per_email(self):
+        '''Returns list of all notifications with same email'''
+        
+        notifications_per_email = Notification.query.filter_by(email=self.email).all()
+        return notifications_per_email
+
+    @classmethod
+    def get_one_notification_id_per_email(cls):
```

Comment:
what's the use case for this? if needed, why not use `first`?

Reply:
This returns a list of one notification id per unique email subscriber. The same email can have multiple rows/multiple notification ids. I implemented this in order to prevent multiple rows with the same email on the front end when being queried for notifications in the get all notifications route.

Reply:
Can you share an example response from the API call? I'm not understanding the reason for this function. We should just query notifications via email. The (marshmallow) schema should handle putting this response in a consumable format for the frontend. 
Realistically, we can just return the exact serialized shape of this Model to the frontend and do a simple change on the frontend to group via email - should be a really simple change.

Reply:
The json I was using was an example before fully defining the models. Feel free to change the frontend handling accordingly.

Reply:
Here is an example: 
<img width="549" alt="Screen Shot 2020-07-01 at 3 00 12 PM" src="https://user-images.githubusercontent.com/42715836/86295555-b6450600-bbab-11ea-91cd-9aedd3dc9508.png">

Reply:
Couldn't fit everything in the screen shot 
`{
    "errors": [],
    "results": {
        "count": 2,
        "data": [
            {
                "email": "test@email.com",
                "notifications": [
                    {
                        "is_active": true,
                        "label": "Load Yellow",
                        "notification_type": "YELLOW_ALARM_LOAD"
                    },
                    {
                        "is_active": true,
                        "label": "Load Red",
                        "notification_type": "RED_ALARM_LOAD"
                    },
                    {
                        "is_active": true,
                        "label": "Price Yellow",
                        "notification_type": "YELLOW_ALARM_PRICE"
                    },
                    {
                        "is_active": true,
                        "label": "Price Red",
                        "notification_type": "RED_ALARM_PRICE"
                    },
                    {
                        "is_active": true,
                        "label": "Price",
                        "notification_type": "PRICE_ALERT"
                    },
                    {
                        "is_active": true,
                        "label": "Import Capacity",
                        "notification_type": "IMPORT_CAPACITY"
                    },
                    {
                        "is_active": true,
                        "label": "Export Capacity",
                        "notification_type": "EXPORT_CAPACITY"
                    },
                    {
                        "is_active": true,
                        "label": "Resource",
                        "notification_type": "RESOURCE_DEPLETION"
                    },
                    {
                        "is_active": false,
                        "label": "Telecomm",
                        "notification_type": "TELECOMM_ALERT"
                    }
                ],
                "pk": 1
            },
            {
                "email": "example@email.com",
                "notifications": [
                    {
                        "is_active": true,
                        "label": "Load Yellow",
                        "notification_type": "YELLOW_ALARM_LOAD"
                    }
                ],
                "pk": 11
            }
        ],
        "detail": ""
    }
}`

Reply:
I will look into moving this onto the frontend !